### PR TITLE
Add remove

### PR DESCRIPTION
**Pull request by aiman-ansari**
user can add new note remove note
user can add add note to archive restore note from archive

### CODE REVIEW

**Review by sanu29 — APPROVED** [review]
Approved. Great work. The UI looks clean also the code is understandable. Remove all the console log before pushing it to production rest all looks great. Great use of ES6... arrow functions used when required also spread very often which is great. Variables are const which again is a good practice.

Good to merge..

File: src/Context/NoteContext.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
+import React from 'react'
+import { createContext, useContext } from "react";
+import { useState } from "react";
+import axios from "axios";
+import { useEffect } from "react";
```
Comment (muskanbagrecha) [inline src/Context/NoteContext.js:5]: You can club this into single import on line 2.

File: src/Context/NoteContext.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+const initialState = {
+    title:"",
+    content:"",
+    priority:"",
+}
+const NoteProvider = ({children}) =>{
+    const [note, setNote] = useState(initialState)
+    const [archive, setArchive] = useState([])
+    const token = localStorage.getItem("token")
```
Comment (muskanbagrecha) [inline src/Context/NoteContext.js:18]: Don't you need to first check if the token exists?

File: src/Context/NoteContext.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    useEffect(()=>{
+        getNote()
+    },[])
+
+    useEffect(()=>{
+        getArchive()
+    },[])
```
Comment (muskanbagrecha) [inline src/Context/NoteContext.js:25]: you can club this in a single effect hook as dependencies are same.

File: src/Context/NoteContext.js [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+            const res = await axios.post('/api/notes', 
+            {
+                note
+            },
+            {
+                headers:{
+                    authorization : token
+            },
+            })
+            console.log("Added:",res.data.notes)
```
Comment (muskanbagrecha) [inline src/Context/NoteContext.js:52]: remove console log

File: src/Context/NoteContext.js [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+                    authorization:token
+                }
+            })
+            setArchive(res.data.archives)
+        }
+        catch(err){
+            console.log(err)
+        }
+    }
+    const addArchieve = async(note) =>{
```
Comment (muskanbagrecha) [inline src/Context/NoteContext.js:106]: please fix the spelling error.

File: src/Pages/Archive/Archive.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import { useNote } from "../../Context/NoteContext"
+import { Header } from "../../Pages/HomePage/Header"
+import { Sidebar } from "../../Pages/HomePage/Sidebar/Sidebar"
+import './Archive.css'
+export const ArchivePage = () =>{
+    const {archive, restoreArchive, addNote, deleteArchive} = useNote()
+    console.log("getting archieves:",archive)
```
Comment (muskanbagrecha) [inline src/Pages/Archive/Archive.js:9]: remove console log

File: src/Pages/Archive/Archive.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import { useNote } from "../../Context/NoteContext"
+import { Header } from "../../Pages/HomePage/Header"
```
Comment (muskanbagrecha) [inline src/Pages/Archive/Archive.js:4]: You can export directly from index file and import it from a single source.

File: src/Pages/Archive/Archive.js [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    return(
+        <div>
+            <Header />
+            <div className="flex">
+                <div className="sidebar">
+                    <Sidebar />
+                </div>
+                <div className="archive-container">
+                    <div>
+                    {archive.map((item) =>
```
Comment (muskanbagrecha) [inline src/Pages/Archive/Archive.js:19]: Can destructure properties.

File: src/Pages/Archive/Archive.js [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+                                onClick={() =>
+                                    deleteArchive(item)
+                                }>
+                                </i>
+                            </span>
+                            <span>
+                                <i className="bi bi-archive-fill" 
+                                    onClick={
+                                        ()=>{
+                                            restoreArchive(item)
```
Comment (muskanbagrecha) [inline src/Pages/Archive/Archive.js:39]: Can extract this into a separate function.

File: src/Pages/Delete/Delete.js [hunk 10]
```diff
@@ -0,0 +1,62 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import { useNote } from "../../Context/NoteContext"
+import { Header } from "../../Pages/HomePage/Header"
+import { Sidebar } from "../../Pages/HomePage/Sidebar/Sidebar"
+// import './Archive.css'
+export const DeletePage = () =>{
+    const {archive, restoreArchive, addNote, deleteArchive , note, deleteNote} = useNote()
+    console.log("getting archieves:",archive)
```
Comment (muskanbagrecha) [inline src/Pages/Delete/Delete.js:9]: remove console log

File: src/Pages/Delete/Delete.js [hunk 11]
```diff
@@ -0,0 +1,62 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import { useNote } from "../../Context/NoteContext"
+import { Header } from "../../Pages/HomePage/Header"
+import { Sidebar } from "../../Pages/HomePage/Sidebar/Sidebar"
+// import './Archive.css'
```
Comment (muskanbagrecha) [inline src/Pages/Delete/Delete.js:6]: remove commented code

File: src/Pages/Delete/Delete.js [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+    return(
+        <div>
+            <Header />
+            <div className="flex">
+                <div className="sidebar">
+                    <Sidebar />
+                </div>
+                <div className="archive-container">
+                    <div>
+                    {note.map((item) =>
```
Comment (muskanbagrecha) [inline src/Pages/Delete/Delete.js:19]: Can destructure here as well

File: src/Pages/Delete/Delete.js [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+                                onClick={() =>
+                                    deleteNote(item)
+                                }>
+                                </i>
+                            </span>
+                            <span>
+                                <i className="bi bi-archive-fill" 
+                                    onClick={
+                                        ()=>{
+                                            // restoreArchive(item)
```
Comment (muskanbagrecha) [inline src/Pages/Delete/Delete.js:39]: Remove commented code.

File: src/Pages/HomePage/MainPage/MainPage.js [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+<div className="note-container">
+            <div >
+                <input type="text" placeholder="Enter your title" className='title-section' />
+            </div>
+            <p>
+                <input type="text" placeholder="Enter your body" className='body-section'></input>
+            </p>
+            <div className='info-section'>
+                <ul className='info-list'>
+                    <li>Created at 14/4/2022</li>
```
Comment (muskanbagrecha) [inline src/Pages/HomePage/MainPage/MainPage.js:16]: Why are you hardcoding the date?

File: src/Pages/HomePage/MainPage/MainPage.js [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+                    <div className='list-icons'>
+                    <li>color</li>
```
Comment (muskanbagrecha) [inline src/Pages/HomePage/MainPage/MainPage.js:18]: I dont understand where this is used.

File: src/Pages/Notes/Edit.js [hunk 16]
```diff
@@ -0,0 +1,90 @@
...
+    }
+    console.log("newnote:",newNote)
+    console.log("from context:",note)
+    const addNotes = () =>{
+        const {title} = newNote
+        console.log(newNote)
+        // addNote(newNote)
+        setNewNote({
+            content:'',
+            "title":newNote.title,
```
Comment (muskanbagrecha) [inline src/Pages/Notes/Edit.js:27]: why is title in string?

File: src/Pages/Notes/Edit.js [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+// import { SingleNote } from "./SingleNote"
+export const EditNote = () =>{
+    const {note, addNote, updateNote} = useNote()
+    const [newNote, setNewNote] = useState(note)
+    const [open , setOpen] = useState(false)
+
+    const handleNotes = (e) =>{
+       setNewNote({...newNote, [e.target.name] : e.target.value})
+    }
+    console.log("newnote:",newNote)
```
Comment (muskanbagrecha) [inline src/Pages/Notes/Edit.js:19]: Remove console log

File: src/Pages/Notes/Edit.js [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+
+    var today = new Date();
+    var date = today.getFullYear()+'-'+(today.getMonth()+1)+'-'+today.getDate();
+    var dateTime = date;
+    console.log(open)
+    return(
+        <div>
+            {open ? 
+            <>
+            <button className='btn btn-primary' onClick={() => setOpen(!open)}>update</button>
```
Comment (muskanbagrecha) [inline src/Pages/Notes/Edit.js:41]: This is not a pure update.

File: src/Pages/Notes/Edit.js [hunk 19]
```diff
@@ -0,0 +1,90 @@
...
+                                <span>
+                                    <i className="bi bi-trash-fill"></i>
+                                </span>
+                                
+                            </div>
+                        </div>
+                    </div>
+            </>:
+                        <button className='btn btn-primary' onClick={() => { addNotes()
+                        setOpen(!open)
```
Comment (muskanbagrecha) [inline src/Pages/Notes/Edit.js:81]: This is not a pure update. Also you can extract this into a separate function.

File: src/Pages/Notes/Note.js [hunk 20]
```diff
@@ -0,0 +1,126 @@
...
+import { SingleNote } from "./SingleNote"
+export const Note = () =>{
+    const {note, addNote} = useNote()
+    const [newNote, setNewNote] = useState(note)
+    const handleNotes = (e) =>{
+       setNewNote({...newNote, [e.target.name] : e.target.value})
+    }
+    console.log(newNote)
+    console.log("from context:",note)
+    const addNotes = () =>{
```
Comment (muskanbagrecha) [inline src/Pages/Notes/Note.js:17]: I think you should make a check that input is not empty.

**Review by muskanbagrecha — APPROVED** [review]
Key pointers:
Remove all console logs, commented code
Try to make your imports cleaner
Try to use prettier for formatting
Extract functions instead of defining them in the jsx itself. 
Make checks such as note should not be added if it is empty
Make your PRs smaller for better review
Good to merge otherwise.  UI looks great!

File: src/Pages/Notes/Edit.js (hunk 16 above)
Comment (aiman-ansari) [reply src/Pages/Notes/Edit.js:27]: that's not part of this pr I was just trying to do edit functionality

File: src/Context/NoteContext.js (hunk 2 above)
Comment (aiman-ansari) [reply src/Context/NoteContext.js:18]: After login then only user can add note I have made different pages for them.

**Review by saurabhchirde — APPROVED** [review]
Couldn't find any issue other than those comments given by @muskanbagrecha .
After resolving them, you can proceed to merge.

**Pull request merged**